wadas/domain/operation_mode.py

```diff
@@ -66,6 +66,64 @@ def init_model(self):
         else:
             logger.debug("Model already initialized, skipping initialization.")
 
+    def _initialize_cameras(self):
+        logger.info("Instantiating cameras...")
+        camera: Camera
+        for camera in cameras:
+            if camera.enabled:
+                if camera.type == Camera.CameraTypes.USB_CAMERA:
+                    # Create thread for motion detection
+                    logger.info("Instantiating thread for camera %s", camera.id)
+                    camera.stop_thread = False
+                    self.camera_thread.append(camera.run())
+                elif (
+                    camera.type == Camera.CameraTypes.FTP_CAMERA
+                    and FTPsServer.ftps_server
+                    and not self.ftp_thread
+                ):
+                    logger.info("Instantiating FTPS server...")
+                    self.ftp_thread = FTPsServer.ftps_server.run()
+        logger.info("Ready for video stream from Camera(s)...")
+
+    def _detect(self, cur_image_path):
+        results, detected_img_path = self.ai_model.process_image(cur_image_path, True)
+        self.last_detection = detected_img_path
+        return results, detected_img_path
+
+    def ftp_camera_exist(self):
+        for camera in cameras:
+            if camera.type == Camera.CameraTypes.FTP_CAMERA:
+                return True
+        return False
+
+    def check_for_termination_requests(self):
+        """Terminate current thread if interrupt request comes from Mainwindow."""
+
+        if self.thread().isInterruptionRequested():
+            logger.info("Request to stop received. Aborting...")
+            # Stop FTPS Server (if running)
+            if self.ftp_camera_exist() and self.ftp_thread and FTPsServer.ftps_server:
+                FTPsServer.ftps_server.server.close_all()
+                FTPsServer.ftps_server.server.close()
+                self.ftp_thread.join()
+            # Stop USB Cameras thread(s), if any.
+            self.process_queue = False
+            for camera in cameras:
+                if camera.type == Camera.CameraTypes.USB_CAMERA:
+                    camera.stop_thread = True
+
+            self.stop_actuator_server()
+
+            self.run_finished.emit()
+            return
+
+    def _initialize_process(self):
+        # Initialize ai model
```

Comment:
we actually do much more than simple Ai model init. please remove comment ad add proper docustring